adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala

```diff
@@ -218,6 +222,31 @@ case class RDDBoundAlignmentRecordRDD private[rdd] (
     import sqlContext.implicits._
     sqlContext.createDataset(rdd.map(AlignmentRecordProduct.fromAvro))
   }
+
+  override def toCoverage(): CoverageRDD = {
+    val covCounts =
+      rdd.rdd
+        .filter(r => {
+          val readMapped = r.getReadMapped
+
+          // validate alignment fields
+          if (readMapped) {
+            require(r.getStart != null && r.getEnd != null && r.getContigName != null,
+              "Read was mapped but was missing alignment start/end/contig (%s).".format(r))
+          }
+
+          readMapped
+        }).flatMap(r => {
+          val t: List[Long] = List.range(r.getStart, r.getEnd)
+          t.map(n => (ReferenceRegion(r.getContigName, n, n + 1), 1))
+        }).reduceByKey(_ + _)
+        .map(r => Coverage(r._1, r._2.toDouble))
+
+    RDDBoundCoverageRDD(covCounts, sequences, None)
+  }
+}
+
+private case class AlignmentWindow(contigName: String, start: Long, end: Long) {
```

Comment:
+1

I like to add an `// empty` comment between the squigglies so that it is obvious they are intentionally empty, but that convention isn't used in this code base.